src/EditorFeatures/Test/Options/GlobalOptionsTests.cs

```diff
@@ -125,10 +126,13 @@ private static Type GetNonNullableType(Type type)
     /// <summary>
     /// Returns another value of the same type that's not equal to the specified <paramref name="value"/>.
     /// </summary>
-    private static object GetNonEqualValue(Type type, object? value)
+    private static object? GetNonEqualValue(Type type, object? value)
     {
         Assert.True(IsOptionValueType(type));
 
+        if (type == typeof(bool?))
+            return value is null ? true : null;
```

Comment:
17.6 branch already handled this case, but the code in 17.6 has diverged greatly from 17.5 in these tests.  so i just did a spot-change to make things work.